v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstobigquery/BigtableChangeStreamsToBQIT.java

```diff
@@ -0,0 +1,520 @@
+
+  private static StaticBigtableResourceManager bigtableResourceManager;
+  private static DefaultBigQueryResourceManager bigQueryResourceManager;
+
+  private JobInfo jobInfo;
+
+  @Test
+  public void testBigtableChangeStreamsToBigQuerySingleMutationE2E() throws IOException {
+    long timeNowMicros = System.currentTimeMillis() * 1000;
+
+    String name = testName.getMethodName();
+    String jobName = createJobName(name);
+    String clusterName = "c1_cluster";
+
+    List<BigtableResourceManagerCluster> clusters = new ArrayList<>();
+    clusters.add(BigtableResourceManagerCluster.create(clusterName, TEST_ZONE, 1, StorageType.HDD));
+
+    bigtableResourceManager.createInstance(clusters);
+
+    BigtableTableSpec cdcTableSpec = new BigtableTableSpec();
+    cdcTableSpec.setCdcEnabled(true);
+    cdcTableSpec.setColumnFamilies(Lists.asList(SOURCE_COLUMN_FAMILY, new String[]{}));
+    bigtableResourceManager.createTable(SOURCE_CDC_TABLE, cdcTableSpec);
+
+    bigtableResourceManager.createAppProfile(
+        APP_PROFILE_ID, true, true, Lists.asList(clusterName, new String[]{}));
+
+    bigQueryResourceManager.createDataset(TEST_REGION);
+
+    LaunchConfig.Builder options =
+        LaunchConfig.builder(jobName, specPath)
+            .addParameter("bigtableTableId", SOURCE_CDC_TABLE)
+            .addParameter("bigtableInstanceId", bigtableResourceManager.getInstanceId())
+            .addParameter("bigtableAppProfileId", APP_PROFILE_ID)
+            .addParameter("bigQueryDataset", bigQueryResourceManager.getDatasetId())
+            .addParameter("bigQueryChangelogTableName", SOURCE_CDC_TABLE + "_changes");
+
+    jobInfo = launchTemplate(options);
+
+    assertThat(jobInfo.state()).isIn(JobState.ACTIVE_STATES);
+
+    String rowkey = UUID.randomUUID().toString();
+    String column = UUID.randomUUID().toString();
+    String value = UUID.randomUUID().toString();
+
+    RowMutation rowMutation =
+        RowMutation.create(SOURCE_CDC_TABLE, rowkey).setCell(SOURCE_COLUMN_FAMILY, column, value);
+    bigtableResourceManager.write(rowMutation);
+
+    String query =
+        String.format(
+            "SELECT * FROM `"
+                + bigQueryResourceManager.getDatasetId()
+                + "."
+                + SOURCE_CDC_TABLE
+                + "_changes`"
+                + " WHERE "
+                + ChangelogColumn.ROW_KEY_STRING.getBqColumnName()
+                + "='%s' AND "
+                + ChangelogColumn.COLUMN_FAMILY.getBqColumnName()
+                + "='%s' AND "
+                + ChangelogColumn.COLUMN.getBqColumnName()
+                + "='%s' AND "
+                + ChangelogColumn.VALUE_STRING.getBqColumnName()
+                + "='%s'",
+            rowkey,
+            SOURCE_COLUMN_FAMILY,
+            column,
+            value);
+
+    Result result =
+        new DataflowOperator(getDataflowClient())
+            .waitForCondition(getWaitForPipelineConfig(jobInfo), dataShownUp(query, 1));
```

Comment:
```suggestion
    Result result =
        pipelineOperator()
            .waitForConditionAndFinish(createConfig(info), dataShownUp(query, 1));
```

Could also use this to make sure rows are written since they will be checked for accuracy after the job runs anyway
```suggestion
    Result result =
        pipelineOperator()
            .waitForConditionAndFinish(
                createConfig(info),
                () -> {
                  return BigQueryRowsCheck.builder(bigQueryResourceManager, table)
                      .setMinRows(1)
                      .build()
                      .get();
                });
```